ramp-engine/ramp_engine/tests/test_dispatcher.py

```diff
@@ -216,3 +231,23 @@ def test_dispatcher_worker_retry(session_toy):
 
     submissions = get_submissions(session_toy, 'iris_test', 'new')
     assert submission_name in [sub[1] for sub in submissions]
+
+
+def test_dispatcher_aws_not_launching(session_toy_aws, caplog):
+    # given the test config file the instance should not be able to launch
+    # due to authentication error
+    # after unsuccessful try the worker should teardown
+    config = read_config(database_config_template())
+    event_config = read_config(ramp_aws_config_template())
+
+    dispatcher = Dispatcher(config=config,
+                            event_config=event_config,
+                            worker=AWSWorker, n_workers=10,
+                            hunger_policy='exit')
+    dispatcher.fetch_from_db(session_toy_aws)
+    dispatcher.launch_workers(session_toy_aws)
+    assert 'AuthFailure' in caplog.text
+    # training should not have started
+    assert 'training' not in caplog.text
+    num_running_workers = dispatcher._processing_worker_queue.qsize()
+    assert num_running_workers == 0
```

Comment:
I suppose there are some subimissions in this db?
Is there a way to make sure that the submission is still in a proper `new` state even though the worker did not start?